PennyMe/MachineChangedRequest.swift

```diff
@@ -0,0 +1,326 @@
+                _selectedSegment = State(initialValue: 0)
+            case "out-of-order":
+                _selectedSegment = State(initialValue: 1)
+            case "retired":
+                _selectedSegment = State(initialValue: 2)
+            default:
+                _selectedSegment = State(initialValue: 0)
+        }
+        // Observe keyboard frame changes
+        keyboardObserver = NotificationCenter.default.publisher(for: UIResponder.keyboardWillChangeFrameNotification)
+                .compactMap { $0.userInfo?["UIKeyboardFrameEndUserInfoKey"] as? CGRect }
+                .map { $0.height }
+                .subscribe(on: DispatchQueue.main)
+                .assign(to: \.keyboardHeight, on: self)
+    }
+    
+    var body: some View {
+        ScrollView{
+        VStack(alignment: .leading, spacing: 5) {
+            // Machine title
+            VStack(alignment: .leading, spacing: 5) {
+                Text("Machine title").foregroundColor(Color.gray) // Label above TextField
+                TextField("Machine title", text: $name)
+                    .textFieldStyle(RoundedBorderTextFieldStyle())
+            }
+            .padding()
+            
+            // adress input field
+            VStack(alignment: .leading, spacing: 5) {
+                Text("Address").foregroundColor(Color.gray) // Label above TextField
+                TextField("Address", text: $address)
+                    .textFieldStyle(RoundedBorderTextFieldStyle())
+            }
+            .padding()
+            
+            // Area input field
+            VStack(alignment: .leading, spacing: 5) {
+                Text("Area").foregroundColor(Color.gray)
+                TextField("Area", text: $area)
+                    .textFieldStyle(RoundedBorderTextFieldStyle())
+            }
+            .padding()
+            
+            // Multimachine input field
+            VStack(alignment: .leading, spacing: 5) {
+                Text("Number of machines").foregroundColor(Color.gray)
+                TextField("Number of machines", text: $multimachine)
+                    .textFieldStyle(RoundedBorderTextFieldStyle())
+            }
+            .padding()
+            
+            // Paywall checkbox
+            Toggle(isOn: $paywall) {
+                            Text("Is there a fee / paywall?").foregroundColor(Color.gray)
+                        }
+                        .padding()
+            
+            // Status segment control
+            VStack(alignment: .leading, spacing: 5) {
+                Text("Machine status").foregroundColor(Color.gray)
+                Picker(selection: $selectedSegment, label: Text("Machine status")) {
+                    Text("Active").tag(0)
+                    Text("Out-of-order").tag(1)
+                    Text("Removed").tag(2)
+                }
+                .pickerStyle(SegmentedPickerStyle())
+            }.padding()
+            // show map
+            Button(action: {
```

Comment:
Button should be centered (like the submit button) and I would suggest to place it at the very top of the view (above address) since it's a very cool and important feature!

Reply:
Also the color can be made more appealing